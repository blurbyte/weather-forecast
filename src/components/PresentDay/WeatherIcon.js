import React from 'react';
import styled from 'styled-components';

import { ClearSkyDay } from '../Icons';

const Wrapper = styled.div`
  width: 11rem;
  height: 11rem;
`;

const WeatherIcon = () => (
  <Wrapper>
    <ClearSkyDay />
  </Wrapper>
);

export default WeatherIcon;
