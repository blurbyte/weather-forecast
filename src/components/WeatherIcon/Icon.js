// Renders appropriate weather conditions icon based on provided weather id

import React from 'react';
import PropTypes from 'prop-types';

import {
  BrokenClouds,
  ClearSkyDay,
  ClearSkyNight,
  FewCloudsDay,
  FewCloudsNight,
  Mist,
  RainDay,
  RainNight,
  ScatteredClouds,
  ShowerRain,
  Snow,
  Thunderstorm
} from '../Icons';

const Icon = ({ isDaytime, weatherId }) => {
  if (weatherId >= 200 && weatherId < 300) {
    return <Thunderstorm />;
  } else if (weatherId >= 300 && weatherId < 400) {
    return <ShowerRain />;
  } else if (weatherId >= 500 && weatherId < 600) {
    if (isDaytime) {
      return <RainDay />;
    }
    return <RainNight />;
  } else if (weatherId >= 600 && weatherId < 700) {
    return <Snow />;
  } else if (weatherId >= 700 && weatherId < 800) {
    return <Mist />;
  } else if (weatherId === 801) {
    if (isDaytime) {
      return <FewCloudsDay />;
    }
    return <FewCloudsNight />;
  } else if (weatherId === 802) {
    return <ScatteredClouds />;
  } else if (weatherId === 803 || weatherId === 804) {
    return <BrokenClouds />;
  } else if (weatherId === 800) {
    if (isDaytime) {
      return <ClearSkyDay />;
    }
    return <ClearSkyNight />;
  } else {
    return <ClearSkyDay />;
  }
};

Icon.propTypes = {
  weatherId: PropTypes.number.isRequired,
  isDaytime: PropTypes.bool
};

export default Icon;
