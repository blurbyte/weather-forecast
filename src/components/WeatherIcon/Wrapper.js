import styled from 'styled-components';

import media from '../../styles/media';

const Wrapper = styled.div`
  width: ${props => (props.small ? '5rem' : '11rem')};
  height: ${props => (props.small ? '5rem' : '11rem')};

  ${media.phone`
    width: ${props => props.small && '3.6rem'};
    height: ${props => props.small && '3.6rem'};
  `};
`;

export default Wrapper;
