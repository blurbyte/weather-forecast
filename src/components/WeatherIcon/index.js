import React from 'react';
import PropTypes from 'prop-types';

import Wrapper from './Wrapper';
import Icon from './Icon';

const WeatherIcon = ({ isDaytime = true, weatherId, small }) => (
  <Wrapper small={small}>
    <Icon isDaytime={isDaytime} weatherId={weatherId} />
  </Wrapper>
);

WeatherIcon.propTypes = {
  weatherId: PropTypes.number.isRequired,
  isDaytime: PropTypes.bool,
  small: PropTypes.bool
};

export default WeatherIcon;
