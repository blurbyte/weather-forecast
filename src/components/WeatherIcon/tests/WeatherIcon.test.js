import React from 'react';
import renderer from 'react-test-renderer';

import WeatherIcon from '../index';

jest.mock('../../Icons', () => ({
  ClearSkyDay: 'ClearSkyDay'
}));

test('renders correctly', () => {
  const component = renderer.create(<WeatherIcon />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders correctly when small prop is provided', () => {
  const component = renderer.create(<WeatherIcon small />);
  expect(component.toJSON()).toMatchSnapshot();
});
