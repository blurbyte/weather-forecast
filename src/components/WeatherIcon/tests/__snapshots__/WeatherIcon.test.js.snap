// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa jdGJAZ"
>
  <Icon
    isDaytime={true}
    weatherId={800}
  />
</div>
`;

exports[`renders correctly when small prop is provided 1`] = `
<div
  className="sc-bdVaJa jmLRzy"
>
  <Icon
    isDaytime={true}
    weatherId={800}
  />
</div>
`;
